### PR TITLE
feat(obstacle_cruise_planner): implemented common obstacle stop

### PR DESCRIPTION
**Pull request by takayuki5168**
## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/utils.hpp [hunk 1]
```diff
@@ -60,7 +60,63 @@ boost::optional<geometry_msgs::msg::Pose> getCurrentObjectPoseFromPredictedPaths
 
 geometry_msgs::msg::Pose getCurrentObjectPose(
   const autoware_auto_perception_msgs::msg::PredictedObject & predicted_object,
-  const rclcpp::Time & obj_base_time, const rclcpp::Time & current_time, const bool use_prediction);
+  const rclcpp::Time & obj_base_time, const rclcpp::Time & current_time, const bool use_predi);
```
Comment (purewater0901) [inline planning/obstacle_cruise_planner/include/obstacle_cruise_planner/utils.hpp:63]: use_prediction

File: planning/obstacle_cruise_planner/src/node.cpp [hunk 2]
```diff
@@ -641,6 +726,43 @@ std::vector<TargetObstacle> ObstacleCruisePlannerNode::filterObstacles(
...
+        if (std::abs(obstacle.velocity) < obstacle_velocity_threshold_from_stop_to_cruise_) {
+          obstacle.has_stopped = true;
+          continue;
+        }
+      } else {
+        if (std::abs(obstacle.velocity) < obstacle_velocity_threshold_from_cruise_to_stop_) {
+          obstacle.has_stopped = true;
+          continue;
+        }
+      }
```
Comment (purewater0901) [inline planning/obstacle_cruise_planner/src/node.cpp:757]: Do we need 'if(has_already_stopped)' ? It seems we always do 'obstacle.has_stopeed=true' regardless of the value has_already stopped

File: planning/obstacle_cruise_planner/src/planner_interface.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    if (closest_behavior_stop_dist_from_ego) {
+      const double closest_obstacle_stop_dist_from_ego = closest_obstacle_dist - dist_to_ego -
+                                                         longitudinal_info_.safe_distance_margin -
+                                                         vehicle_info_.max_longitudinal_offset_m;
+
+      const double stop_dist_diff =
+        *closest_behavior_stop_dist_from_ego - closest_obstacle_stop_dist_from_ego;
+      std::cerr << stop_dist_diff << " " << *closest_behavior_stop_dist_from_ego << " "
+                << closest_obstacle_stop_dist_from_ego << std::endl;
```
Comment (purewater0901) [inline planning/obstacle_cruise_planner/src/planner_interface.cpp:128]: Remove

Comment (takayuki5168) [reply planning/obstacle_cruise_planner/src/planner_interface.cpp]: https://github.com/autowarefoundation/autoware.universe/pull/1185/commits/ac6019cc49486c2b375df75ab6b84218333527b1
DONE

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/utils.hpp (hunk 1 above)
Comment (takayuki5168) [reply planning/obstacle_cruise_planner/include/obstacle_cruise_planner/utils.hpp]: https://github.com/autowarefoundation/autoware.universe/pull/1185/commits/6b8a13984565319e15bb2dae04c9434dc9f22d69
DONE

File: planning/obstacle_cruise_planner/src/node.cpp (hunk 2 above)
Comment (purewater0901) [reply planning/obstacle_cruise_planner/src/node.cpp:757]: done

**Review by purewater0901 — APPROVED** [review]

**Pull request merged**